static/styles/zulip.scss

```diff
@@ -3034,3 +3034,84 @@ div.message_content button.tictactoe-square:disabled {
 .flatpickr-months .numInputWrapper span {
     display: none;
 }
+
+i.fa-smile-o span {
+    display: none;
+}
+
+i.fa-chevron-down span {
+    display: none;
+}
+
+.star-message span {
+    display: none;
+}
+
+i.fa-smile-o:hover span {
+    -webkit-border-radius: 4px;
+    -moz-border-radius: 4px;
+    border-radius: 4px;
+    background-color: hsla(0, 0%, 7%, 0.8);
+    padding: 3px 5px;
+    font-size: 11px;
+    line-height: 1.3;
+    opacity: 1;
+    display: block;
+    z-index: 9999;
+    color: hsla(360, 100%, 100%, 1);
+    text-align: center;
+    text-decoration: none;
+    left: -60px;
+    margin: 10px;
+    width: 100px;
+    position: absolute;
+    top: 10px;
+    font-family: "Helvetica Neue", Arial, Helvetica, sans-serif;
+    text-decoration: none !important;
+}
+
+i.fa-chevron-down:hover span {
+    -webkit-border-radius: 4px;
+    -moz-border-radius: 4px;
+    border-radius: 4px;
+    background-color: hsla(0, 0%, 7%, 0.8);
+    padding: 3px 5px;
+    font-size: 11px;
+    line-height: 1.3;
+    opacity: 1;
+    display: block;
+    z-index: 9999;
+    color: hsla(360, 100%, 100%, 1);
+    text-align: center;
+    text-decoration: none;
+    left: -60px;
+    margin: 10px;
+    width: 100px;
+    position: absolute;
+    top: 10px;
+    font-family: "Helvetica Neue", Arial, Helvetica, sans-serif;
+    text-decoration: none !important;
+}
+
+.star-message:hover span {
+    -webkit-border-radius: 4px;
+    -moz-border-radius: 4px;
+    border-radius: 4px;
+    background-color: hsla(0, 0%, 7%, 0.8);
+    padding: 3px 5px;
+    font-size: 11px;
+    line-height: 1.3;
+    opacity: 1;
+    display: block;
+    z-index: 9999;
+    color: hsla(360, 100%, 100%, 1);
+    text-align: center;
+    text-decoration: none;
+    left: -30px;
+    margin: 10px;
+    width: 100px;
+    position: absolute;
+    top: 10px;
+    font-family: "Helvetica Neue", Arial, Helvetica, sans-serif;
+    text-decoration: none !important;
+}
```

Comment:
These CSS rules are poorly constrained (potentially affecting a bunch of elements unexpectedly) and repetitive; I think it'd be better to have this behavior be controlled by some sort of `tooltip` class that we can add to elements that want this behavior.

Reply:
Thanks for reviewing. I would make the necessary changes.